solidity/ts/testsuite/simulator/utils/utilities.ts

```diff
@@ -177,12 +176,63 @@ export const mintCash = async (mockWindowEthereum: MockWindowEthereum, mintAmoun
 	await mockWindowEthereum.addStateOverrides({ [cashAddress]: { stateDiff: stateSets }})
 }
 
+export const mintRep = async (mockWindowEthereum: MockWindowEthereum, mintAmounts: { address: Address, amount: bigint }[]) => {
+	const repAddress = await getRepAddress(createWriteClient(mockWindowEthereum, VITALIK, 0))
+	const overrides = mintAmounts.map((mintAmount) => {
+		const encodedKeySlotHash = keccak256(encodeAbiParameters([{ type: 'address' }, { type: 'uint256' }], [mintAmount.address, 1n]))
+		return { key: encodedKeySlotHash, value: mintAmount.amount }
+	})
+	const stateSets = overrides.reduce((acc, current) => {
+		acc[current.key] = current.value
+		return acc
+	}, {} as { [key: string]: bigint } )
+	await mockWindowEthereum.addStateOverrides({ [repAddress]: { stateDiff: stateSets }})
+}
+
+export const getCashAddress = async (client: ReadClient) => {
+	return await client.readContract({
+		abi: ABIS.mainnet.universe,
+		functionName: 'cash',
+		address: addressString(AUGUR_UNIVERSE_ADDRESS),
+		args: []
+	})
+}
+
+export const getRepAddress = async (client: ReadClient) => {
+	return await client.readContract({
+		abi: ABIS.mainnet.universe,
+		functionName: 'getReputationToken',
+		address: addressString(AUGUR_UNIVERSE_ADDRESS),
+		args: []
+	})
+}
+
 export const setupTestAccounts = async (mockWindowEthereum: MockWindowEthereum) => {
 	const accountValues = TEST_ADDRESSES.map((address) => {
 		return { address: addressString(address), amount: 1000000n * 10n**18n}
 	})
 	await mintETH(mockWindowEthereum, accountValues)
 	await mintCash(mockWindowEthereum, accountValues)
+	await mintRep(mockWindowEthereum, accountValues)
+}
+
+export const deployAugurMarket = async (client: WriteClient): Promise<Address> => {
+	await approveCash(client, addressString(AUGUR_ADDRESS))
+	const endTime = BigInt(Math.floor(Date.now() / 1000) + 100000)
+	const blockNumber = await client.getBlockNumber()
+	await client.writeContract({
+		abi: ABIS.mainnet.universe,
+		functionName: 'createYesNoMarket',
+		address: addressString(AUGUR_UNIVERSE_ADDRESS),
+		args: [endTime, 0n, addressString(NULL_ADDRESS), 0n, client.account.address, "{}"]
+	})
+	const logs = await client.getLogs({
+		address: addressString(AUGUR_ADDRESS),
+		event: parseAbiItem("event MarketCreated(address indexed universe, uint256 endTime, string extraInfo, address market, address indexed marketCreator, address designatedReporter, uint256 feePerCashInAttoCash, int256[] prices, uint8 marketType, uint256 numTicks, bytes32[] outcomes, uint256 noShowBond, uint256 timestamp)"),
```

Comment:
I think we add ABI of the Augur Constant Product Market and get this information from it, then you also get type safety

Reply:
this provides type safety already

Reply:
I do find it convenient to just have a single `const ABI = [ ... ] as const` that we can just reference, rather than having to type out the function definition in canonical form everywhere.